src/app/shared/table/table.component.ts

```diff
@@ -128,8 +129,30 @@ export class TableComponent implements OnInit, AfterViewInit, OnDestroy {
     }
   }
 
-  public filterChanged(filterDef: TableFilterDef) {
-    this.dataSource.filterChanged(filterDef);
+  public filterChanged(filterDef: TableFilterDef, key?: string) {
+    if(filterDef.id === 'date_menu_dropdown') {
```

Comment:
This code should not be there.
The two date filters should be passed to the DateMenuDropdownFilter object which will update the Json date objects according the drop down selection.
The this code should be in the DateMenuDropdownFilter.
I don't know if the framework is supporting that currently.
You can check with Mahdi who did the dependencies between filters.